lib: location: Remove support for HERE location services (#21262)
nrfconnect/sdk-nrf: doc/nrf/releases_and_maturity/migration/migration_guide_3.0.rst

   * You must set the :kconfig:option:`CONFIG_NRF_CLOUD` Kconfig option to access the nRF Cloud libraries.
     This option is now disabled by default to prevent the unintended inclusion of nRF Cloud Kconfig variables in non-nRF Cloud projects, addressing a previous issue.

Location library
----------------

.. toggle::

   For applications and samples using the :ref:`lib_location` library:

   * Support for HERE location services and the :kconfig:option:`CONFIG_LOCATION_SERVICE_HERE` have been removed.
     To use external location services, enable the :kconfig:option:`CONFIG_LOCATION_SERVICE_EXTERNAL` option and implement the required APIs.

>    * The ``service`` parameter in ``struct location_cellular_config`` and ``struct location_wifi_config`` have been removed.

Reviewer: ```suggestion
   * The ``service`` parameter in :c:struct:`location_cellular_config` and :c:struct:`location_wifi_config` has been removed.
```

>      Only one location service is supported by the library, so the ``service`` parameter is no longer needed.

Reviewer: ```suggestion
     The library supports only one location service, so the ``service`` parameter is no longer needed.
```
